### ISSUE TITLE
thread 'main' panicked at 'Unexpected type of Aux for phaseset

### ISSUE THREAD

**Issue by jmcdani** [wdecoster/cramino#24]
Hi, 

I'm trying to run cramino for a phased .bam that was generated from PacBio Revio data using PacBio HiPhase. When I used the `--phased`  option I get the following error
```
thread 'main' panicked at 'Unexpected type of Aux for phaseset: I32(11863)', src/extract_from_bam.rs:171:18
stack backtrace:
   0: rust_begin_unwind
             at /rustc/d5c2e9c342b358556da91d61ed4133f6f50fc0c3/library/std/src/panicking.rs:593:5
   1: core::panicking::panic_fmt
             at /rustc/d5c2e9c342b358556da91d61ed4133f6f50fc0c3/library/core/src/panicking.rs:67:14
   2: cramino::extract_from_bam::extract
   3: cramino::main
```
If I remove `--phased`, cramino runs.  I'm wondering if somehow the phaseset ID is inconsistent with what cramino is expecting.  When I look at representative phase set IDs in .bam it looks something like this: `PS:i:11863`.    Could this be an issue with integer type I32 vs U32?  Seems to think this is I32 however looking in your code looks like its expected U32, which I thought "11863" would be. 
```
fn get_phaseset(record: &bam::Record) -> Option<u32> {
    match record.aux(b"PS") {
        Ok(value) => match value {
            Aux::U8(v) => Some(u32::from(v)),
            Aux::U16(v) => Some(u32::from(v)),
            Aux::U32(v) => Some(v),
            _ => panic!("Unexpected type of Aux for phaseset: {:?}", value),
        },
        Err(_e) => None,
    }
}
```
 Thank you !

**Comment by wdecoster** [comment]
Thank you for reporting this! I'm not entirely sure what it means, and how the type of the tags is determined to be one of the numeric types, but I know how to fix it. I'll make a new release soon.

**Comment by jmcdani** [comment]
Thank you Wouter for addressing that so quickly!  Phasing option is working now.

**Comment by wdecoster** [comment]
Ah, then I will also fix it for I16 :-D 

New release https://github.com/wdecoster/cramino/releases/tag/v0.14.4

**Comment by asvarvara** [comment]
I am simply running cramino without the --phased flag and it throws me this error:

thread 'main' panicked at src/extract_from_bam.rs:180:18:
Unexpected type of Aux for NM tag: I8(1)

I am using a dorado bam with modified bases and cramino version 0.14.4

**Comment by wdecoster** [comment]
Aha, another unexpected Aux value. I will fix that one too :)

**Comment by asvarvara** [comment]
Thanks a lot :) I will patiently wait the next release.

**Comment by wdecoster** [comment]
Should be solved in [v0.14.5](https://github.com/wdecoster/cramino/releases/tag/v0.14.5)

**Comment by asvarvara** [comment]
> Should be solved in [v0.14.5](https://github.com/wdecoster/cramino/releases/tag/v0.14.5)

Thanks a lot, I can confirm it has been fixed!

**Comment by wdecoster** [comment]
Good to hear!
